File: csharp/CsFind/CsFindLib/EmbeddedResource.cs
using System;
using System.IO;
using System.Reflection;
using System.Text;

namespace CsFindLib
{
    public static class EmbeddedResource
    {
        public static string GetResourceFileContents(string namespaceAndFileName)
        {
            try
            {
                using var stream = Assembly.GetAssembly(typeof(EmbeddedResource))!.GetManifestResourceStream(namespaceAndFileName);
                using var reader = new StreamReader(stream!, Encoding.UTF8);
                return reader.ReadToEnd();
            }
            catch(FileNotFoundException)
            {
                throw new Exception($"Failed to read Embedded Resource {namespaceAndFileName}");
            }
        }
    }
}

File: fsharp/FsFind/FsFindLib/EmbeddedResource.fs
namespace FsFindLib

open System.IO
open System.Reflection

module EmbeddedResource = 

    type T = interface end

    let GetResourceFileContents (namespaceAndFileName : string) : string = 
        let contents =
            try
                use stream = Assembly.GetAssembly(typeof<T>).GetManifestResourceStream(namespaceAndFileName)
                use reader = new StreamReader(stream)
                reader.ReadToEnd()
            with
            | :? IOException as e -> printfn $"%s{e.Message}"; ""
        contents
